### ISSUE TITLE
SwaggerSpecFilter isParamAllowed is cached, killing swagger.json

### ISSUE THREAD

**Issue by swartzrock** [swagger-api/swagger-core#1742]
Using swagger-core and swagger-jaxrs 1.5.8

Steps:
1) Define your own SwaggerSpecFilter and require an api_key for isPropertyAllowed
2) Start your server and access swagger
Result: invalid api key, nothing shown
3) Now enter your api_key
Expected Result: the full swagger.json including all properties
Actual Result: the swagger.json will not include any properties, as they were not available the first time that swagger.json was accessed.

**Comment by webron** [comment]
Thanks for reporting. Version of swagger-core?

**Comment by swartzrock** [comment]
Using 1.5.8. Also updated the description.

**Comment by nightkc** [comment]
Hi. I also encountered the same issue,
I took a look at the functions, and found the cause in side the method: **io.swagger.core.filter.SpecFilter.filterDefinitions(...)**

clonedModel.getProperties() returns Map<String, Property> which was referencing to the statically shared swagger definitions before filtering. The code below is taken from the method:
`
            Model clonedModel = (Model) definition.clone();
            if (clonedModel.getProperties() != null) {
                clonedModel.getProperties().clear();
            }
            if( definition.getVendorExtensions() != null && clonedModel.getVendorExtensions() != null ){
                clonedModel.getVendorExtensions().putAll( definition.getVendorExtensions());
            }

            clonedModel.setProperties(clonedProperties);
            clonedDefinitions.put(key, clonedModel);
            }
`
I think the clonedModel.getProperties().clear(); was called to fix bugs in setProperties which clonedModel.setProperties calls addProperty in loop (As a result, clonedModel has all properties before filtering, and clonedModel.setProperties(clonedProperties) does "add" instead of "set" ). However, Model clonedModel = (Model) definition.clone() cloned the original reference into the clone obj, so that when clonedModel.getProperties().clear() is called, the original properties (none-filtered cache shared by all threads) get erased, and overwritten by clonedModel.setProperties(...) .
